src/librustdoc/html/static/source-script.js

```diff
@@ -140,4 +140,9 @@ function createSourceSidebar() {
     });
 
     main.insertBefore(sidebar, main.firstChild);
+    // Focus on the current file in the source files sidebar.
+    var selected_elems = Array.prototype.slice.call(sidebar.getElementsByClassName("selected"));
```

Comment:
Surely `getElementsByClassName` is already array-like because it's a live HTML collection.  So why bother with this `slice()` invocation?

Reply:
A performance concern: chrome retriggers the whole selector everytime you try to access an element whereas firefox handles it nicely. It was the main reason why the rust docs were so slow on chrome a year ago. So since then, I try to pay attention to it.

Reply:
I see, something like:
```
const focus_me = (sidebar.getElementsByClassName("selected"))[0];
if (focus_me) {
    focus_me.focus()
}
```

would also work, but I don't mind if this use of slice is common in the codebase.

Reply:
Everytime we call `onEachLazy`, it's doing it. And the problem is getting access to the nth element, not `getElementsByClassName` itself, so I think the issue would be the same.

Reply:
So your approach is "make an array by iterating the full html collection, then indexing [0] out of it is cheap" where my suggestion is "just get the zeroth element of the html collection and then check if it's defined"  Both index the collection exactly once, though your approach indexes the entire collection (likely small) where mine explicitly only indexes the first element out.

Reply:
In this case, the whole array is either one element or zero. It just seemed better to do it this way because otherwise you have to query it twice (the first time to check the length and the second to focus on the first element).

Reply:
You don't have to query it twice.  Simply take the zeroth element of the collection (one query) and if it's not `undefined` then it's there and you can run it.  Think of my suggestion as:

```rust
if let Some(focus_me) = sidebar.getElementsByClassName("selected").next() {
    focus_me.focus();
}
```

However, as I said, if you're using the `Array.prototype.slice.call()` approach elsewhere then it's "idiom" for this codebase and I'm okay with it.  Whether you choose to change or not, I'm okay with the approach.  I'm assuming you've confirmed it works, I'm on the wrong computer to do a rustdoc build right now.

Reply:
I thought it would crash in such case. Damn I hate JS... Updating then, thanks for the tip!

Reply:
Up to you.  As is is okay as I said.  Either way, you can r=kinnison assuming you've checked it does what you expect in a browser :D

Reply:
Just checked and it works as expected. Dark JS magic is back! Thanks for the tip!